### PR TITLE
Support both guzzle 5 and guzzle 6

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Ping @mtdowling, @jeremeamia, @lucasmichot.

### CODE REVIEW

File: src/Illuminate/Mail/Transport/MailgunTransport.php [hunk 1]
```diff
@@ -79,12 +80,21 @@ public function stop()
 	 */
 	public function send(Swift_Mime_Message $message, &$failedRecipients = null)
 	{
-		return $this->client->post($this->url, ['auth' => ['api', $this->key],
-			'multipart' => [
+		$options = ['auth' => ['api', $this->key]];
+
+		if (version_compare('6', $client::VERSION) === 1) {
```
Comment (jeremeamia) [inline src/Illuminate/Mail/Transport/MailgunTransport.php]: `$this->client`?

File: src/Illuminate/Mail/Transport/MandrillTransport.php [hunk 2]
```diff
@@ -63,14 +63,20 @@ public function stop()
...
-			],
-		]);
+		$data = [
+			'key' => $this->key,
+			'to' => $this->getToAddresses($message),
+			'raw_message' => (string) $message,
+			'async' => false,
+		];
+
+		if (version_compare('6', $client::VERSION) === 1) {
```
Comment (jeremeamia) [inline src/Illuminate/Mail/Transport/MandrillTransport.php]: `$this->client` again. Sorry, I mislead you with my comment on the other PR.

**Comment by jeremeamia** [comment]
Looks good conceptually! Need to the correct the `$client` references, and I think there was a test in the other PR that might need to be updated as well.

**Comment by lucasmichot** [comment]
TBO, I don't see the real added value of Guzzle 6
What postfile brings more compared to v5 ?

File: src/Illuminate/Mail/Transport/MandrillTransport.php (hunk 2 above)
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MandrillTransport.php]: PHP also doesn't like that. I can't put the `::` there. You get a parse error. It also doesn't help that I forgot to require guzzle in composer too...

**Comment by GrahamCampbell** [comment]
Gah. How about we just delete this test? Ping @taylorotwell.

**Comment by GrahamCampbell** [comment]
Alight. Manually tested this for all 4 cases, and it works just fine. :)

**Pull request merged**